UI Refactor `[AARD-1903]`

PepperLola opened the pull request:
## UI Refactor

AARD-1903

Refactored UI to improve the process of creating and using modals and panels. Rather than registering each possible UI element and then opening them with hardcoded IDs, the UI element itself is now passed into the relevant `open*` functions. In a `tsx` file, this looks something like `openPanel(<ConfigurePanel />)` but in regular `.ts` files it would look like `openPanel(React.createElement(ConfigurePanel))`.

<details>
    <summary>Panels vs Modals</summary>

There are two types of screens: Panels and Modals. I decided to generalize these as UI screens. There can only ever be one modal open at a time, and the modals demand attention from the user, meaning they block interaction to everything else behind them until they're dismissed. There can be any number of panels open and are draggable, so they're intended to be passive and able to remain open on the side while using Synthesis. For example, the main menu is a modal because we need the user to choose a mode before we can load a scene. The configuration panel, on the other hand, doesn't require the attention of the user so it can be made a panel. It's my personal opinion that any screen that doesn't *necessarily* require the attention of the user should be made a panel, as a modal can disrupt a user's focus.

The bulk of the UI management happens in [UIProvider](https://github.com/Autodesk/synthesis/blob/jwrigh/1903/ui-refactor-tmp/fission/src/ui/UIProvider.tsx). The screen type looks as follows:
```ts
interface UIScreen<T> {
    id: string
    parent?: UIScreen<unknown>
    content: ReactElement
    props: ModalProps | PanelProps
    onClose: UICallback<[CloseType], void>
    onCancel: UICallback<[], void>
    onAccept: UICallback<[T], void>
    onBeforeAccept: UICallback<[], T>
}
```

The `id`s are automatically generated when opening a screen and returned from the open function. They're required for closing a panel, which means you can only close a screen that you open (or if the code that opened the screen passes the `id` down elsewhere). This was a byproduct of the way I designed the opening (i.e. passing in a react component rather than an `id` so you don't have to register screens beforehand) but I think also makes sense as a requirement, as our previous code frequently closed panels in random places in a way that made the flow difficult to follow.

The `parent` is optional, but is a screen object referencing the screen that opened the current one, and comes from an optional argument in the open functions specifying the parent. This allows a screen to close its parent, as it has direct access to the `id`.

The `content` is just a `ReactElement` and is generally going to be a functional component that we define. This is what is rendered on the screen.

The `props` are either Modal- or Panel-specific properties. Each `Modal` and `Panel` interface that extends `UIScreen<T>` overwrite the `props` field with one of the relevant type. The properties interfaces are as follows:

```ts
interface UIScreenProps {
    configured: boolean
    title?: string
    hideCancel?: boolean
    hideAccept?: boolean
    cancelText?: string
    acceptText?: string
}

interface ModalProps extends UIScreenProps {
    // required for PanelProps to not satisfy ModalProps
    type: "modal"
    allowClickAway?: boolean
}

interface PanelProps extends UIScreenProps {
    type: "panel"
    position: PanelPosition
}
```

The `configured` property can be ignored, and is only used internally to denote whether the screen has been configured; the screen will only be rendered when once it has been configured. The `title` will render at the top of the screen. The `hideCancel` and `hideAccept` props will disable rendering of the relevant buttons when true, and `cancelText` and `acceptText` will change the text shown in the buttons. Each modal has a clickaway listener that will close it if the user clicks the background, but this behavior can be disabled by setting the `allowClickAway` property to `false`. As panels can exist alongside each other and can be moved, they also have configurable opening positions, which can be configured by setting the `position` property.

As the `ModalProps` interface only contains one optional field, there's nothing stopping a `PanelProps` instance from satisfying `ModalProps`, which is why the `type` fields were added. They can also be ignored and only serve the purpose of allowing the enforcement of the right properties type for each type of screen.
</details>

<details>
<summary>Callbacks</summary>

I created a `UICallback` object that can contain a default function and a user-defined function (or any combination of the two, as they are both optional). The user-defined function always runs first when defined, and the return value of the entire callback will always be that of the user-defined when it's not undefined. The user-defined functions are passed into the screen open functions, and the default functions are provided when calling `configureScreen` from within a screen element. Each screen has the following callbacks:

```ts
interface UIScreenCallbacks<T> {
    onClose?: (closeType: CloseType) => void
    onCancel?: () => void
    onBeforeAccept?: () => T
    onAccept?: (arg: T) => void
}
```

`onClose` is always called before any other callback when a screen is closed. The callback has a parameter of the `CloseType`, which can either be `Accept`, `Cancel`, or `Overwrite`. They are pretty self-explanatory - `Accept` and `Cancel` depend on the button that was pressed, and `Overwrite` is used when one screen takes the place of another. When a modal is opened, any previously open modal will also be closed and its `onClose` function will be called with the `Overwrite` close type. `onCancel` is called when the cancel button is clicked (or whenever a function is called to close a screen with the intent of canceling its purpose). `onBeforeAccept` is called, as the name suggests, before `onAccept`. It is only configurable from within a screen and can thus only contain a default implementation (passed in when calling `configureScreen`). It can return a value of type `T` (defined when creating a screen) that is then passed as an argument to `onAccept`, which can only contain a user-defined value configured in the open function for the screen.
</details>

<details>
    <summary>UIProvider and UIRenderer</summary>
    [UIProvider](https://github.com/Autodesk/synthesis/blob/jwrigh/1903/ui-refactor-tmp/fission/src/ui/UIProvider.tsx) exposes the current modal and panels, along with all the relevant ui-related functions:

```ts
type UIContextProps = {
    modal?: Modal<unknown>
    panels: Panel<unknown>[]
    openModal: OpenModalFn
    openPanel: OpenPanelFn
    closeModal: CloseModalFn
    closePanel: ClosePanelFn
    addToast: AddToastFn
    configureScreen: ConfigureScreenFn
}
```
These can be imported by calling the `useUIContext` hook from within a React component (e.g. within a panel or modal).

[UIRenderer](https://github.com/Autodesk/synthesis/blob/jwrigh/1903/ui-refactor-tmp/fission/src/ui/UIRenderer.tsx) just renders the current modal and panels, passing through the props and parent.
</details>

<details>
    <summary>Opening UI Screens</summary>

UI Screens can be opened using their relevant open functions, with the following type signatures:

```ts
type OpenModalFn = <T>(
    contents: ReactElement,
    parent?: UIScreen<T>,
    props?: Omit<ModalProps, "type" | "configured">
) => string
type OpenPanelFn = <T>(
    contents: ReactElement,
    parent?: UIScreen<T>,
    props?: Omit<PanelProps, "type" | "configured">
) => string
```
To open a screen, you need to call the relevant open function and pass in a `ReactElement`. You can optionally pass in a parent (which is where the parent property from before comes from) and properties you want to configure (which is where the user-defined callbacks come from). The functions will return the randomly-generated `id` for the new screen.
</details>

<details>
    <summary>Closing UI Screens</summary>

To close a screen, you need to first have the `id` of the screen you want to close, which will have previously been returned from the function that opened that screen. To close the screen, you simply call the relevant close function. As there can only ever be one modal open at once, the `closeModal` function only requires a close type to be passed in, which was defined above to be one of `Accept`, `Cancel`, or `Overwrite`. The `closePanel` function requires the `id` of the panel to close along with the close type.
</details>

<details>
    <summary>Rendering Toasts</summary>

I installed the `notistack` package to add a notification stack to the bottom-left corner of the screen like we had previously. UIProvider exposes an `addToast` function that takes in a type (from `notistack`) along with any number of string arguments beyond that, which will be joined by line breaks and rendered in the notification.
</details>

<details>
    <summary>Global Functions</summary>

The [GlobalUIControls](https://github.com/Autodesk/synthesis/blob/jwrigh/1903/ui-refactor-tmp/fission/src/ui/components/GlobalUIControls.ts) file exports `globalAddToast`, `globalOpenPanel` and `globalOpenModal` which are all set to the respective functions from UIProvider. This is so that non-React-component code can perform any of these actions. This is especially important for adding toasts, as there are many places we would want to provide feedback to the user (e.g. in the event of an error) from code that isn't in a React component.
</details>

<details>
    <summary>Creating a UI Screen</summary>

Creating a new screen is as simple as creating a new file with a functional component in the correct directory (i.e. `src/ui/modals/` for modals, `src/ui/panels/` for panels) and exporting it. The type of the component must be either `PanelImplProps<T>` or `ModalImplProps<T>`, where `T` is the type of the data to be returned from `onBeforeAccept` and passed into `onAccept`.

Those types allow the screens to access their object (`Modal` or `Panel`, the interface introduced earlier with the `id`, `parent`, etc.) along with that of their parent as props.

Each screen must also be configured, which must be done via the `configureScreen` function exposed by the UI context within UIProvider. **This is necessary for all panels**, even if you don't plan to override any properties or callbacks. If this isn't done, the UI renderer won't consider the screen to have been configured and as such will not render it.

    The type signature for the configure function is a bit difficult to understand at first:
```ts
type ConfigureScreenFn = <T extends UIScreen<any>>(
    screen: T,
    props: T extends Panel<infer _> ? Partial<Omit<PanelProps, "configured">> : Partial<Omit<ModalProps, "configured">>,
    callbacks: T extends Modal<infer S>
        ? Omit<Partial<UIScreenCallbacks<S>>, "onAccept">
        : T extends Panel<infer S>
          ? Omit<Partial<UIScreenCallbacks<S>>, "onAccept">
          : never
) => void
```

It essentially takes either a panel or a modal as its first argument and, depending on which is provided, will require the props for that type to be passed as the second argument (guarding against configuring a modal with panel props). The callbacks provided here to the `callbacks` argument are then set as the default implementation for the callbacks (as opposed to the user-defined implementation; discussed above).

The APS management modal is a good, small example of creating a modal:
```ts
import { Stack } from "@mui/material"
import type React from "react"
import { useEffect, useState } from "react"
import { HiUser } from "react-icons/hi"
import APS from "@/aps/APS"
import type { ModalImplProps } from "@/ui/components/Modal"
import { useUIContext } from "../helpers/UIProviderHelpers"

const APSManagementModal: React.FC<ModalImplProps<void>> = ({ modal }) => {
    const { configureScreen } = useUIContext()
    const [userInfo, _] = useState(APS.userInfo)
    useEffect(() => {
        const onBeforeAccept = () => {
            APS.logout()
        }

        configureScreen(modal!, { title: userInfo?.name ?? "Not signed in", acceptText: "Logout" }, { onBeforeAccept })
    }, [modal, userInfo?.name])

    return (
        <Stack spacing={10} direction="row">
            {userInfo?.picture ? (
                <img alt={userInfo?.name} src={userInfo?.picture} className="h-10 rounded-full" />
            ) : (
                <HiUser />
            )}
        </Stack>
    )
}

export default APSManagementModal
```

We can see that it imported the `configureScreen` function using the `useUIContext` hook. This is also how to import the `openModal`, `closeModal`, `openPanel`, and `closePanel` functions.

It also called `configureScreen` within a `useEffect`, which is the preferred way of configuring a screen. Even if no configuration were necessary, it would still need to call `configureScreen` with empty property objects, as follows:

```ts
configureScreen(modal!, {}, {})
```

This modal could then be opened in another component like so:
```ts
// first import the openModal function
const { openModal } = useUIContext()

// then open the modal
openModal(<APSManagementModal />)
```

or from a regular TypeScript file, where you can't create a JSX/TSX element:
```ts
// first import React and the globalOpenModal function
import React from "react"
import { globalOpenModal } from "@/ui/components/GlobalUIControls"

// then open the modal
globalOpenModal(React.createElement(APSManagementModal))
```
</details>

<details>
    <summary>MUI</summary>

I migrated (almost?) everything to use [MUI Material UI](https://mui.com/material-ui/) instead of our own built-in components. Some more complex components were kept but internally use MUI components (e.g. SelectMenu), but some were discarded entirely (e.g. removed Dropdown in favor of MUI's Select).

The most important components going forward are `Stack` (flex box), `Box` (general container), `Select` (dropdown menu), `Button` (button), and `TextField` (text input field). A full list of MUI components can be found [here](https://mui.com/material-ui/all-components/).
</details>

---

Before merging, ensure the following criteria are met:

- [ ] All acceptance criteria outlined in the ticket are met.
- [ ] Necessary test cases have been added and updated.
- [ ] A feature toggle or safe disable path has been added (if applicable).
- [ ] User-facing polish:
  - Ask: *"Is this ready-looking?"*
- [x] Cross-linking between Jira and GitHub:
  - PR links to the relevant Jira issue.
  - Jira ticket has a comment referencing this PR.

PepperLola committed:
refactor: removed old UI and rewrote UIProvider and UIRenderer

PepperLola committed:
refactor: progress on new modal component

PepperLola committed:
refactor: added draggable panel with configurable starting locations

PepperLola committed:
refactor: added toasts and started on theming

PepperLola committed:
refactor: reimplemented panels

PepperLola committed:
chore: satisfy eslint and fix config panel appearance and open/close functionality

PepperLola committed:
fix: properly accept toast variants, fixed value -> children for some buttons, made buttons contained by default

PepperLola committed:
refactor: added back ws view panel

PepperLola committed:
chore: format and improve styled button utils

PepperLola committed:
fix: theme is editable again and can be randomized

PepperLola committed:
refactor: started on modals

PepperLola committed:
refactor: more modals

PepperLola committed:
refactor: added onBeforeAccept callback and updated types

PepperLola committed:
fix: add back scrollview

PepperLola committed:
feat: configurable cancel and accept button text

PepperLola committed:
refactor: create hooks for using various states

PepperLola committed:
fix: use mui components properly

PepperLola committed:
refactor: ui screens now have sets of callbacks so they aren't overwritten

PepperLola committed:
refactor: only have two possible callbacks, don't allow onAccept user definition

PepperLola committed:
refactor: add configureScreen utility function to UIContext

PepperLola committed:
fix: cleanup and omit type field from open screen function props

PepperLola committed:
chore: cleanup and fix build issues

PepperLola committed:
chore: format and delete old modal/panel helpers

PepperLola committed:
fix: instantiate onAccept UICallback on screen open

PepperLola committed:
refactor: use StatefulSlider component for all sliders

PepperLola committed:
fix: don't allow panel drag on sliders (<span>)

PepperLola committed:
refactor(typography): create label component with size presets

PepperLola committed:
chore: fix props for all screens

TODO: figure out a better way to handle prop change in child

PepperLola committed:
fix: don't allow import if already importing

PepperLola committed:
refactor: added tooltip option to slider and renamed

PepperLola committed:
feat: add circular progress to aps loading in import panel

PepperLola committed:
fix: don't render view cube when main menu open

PepperLola committed:
fix: toasts render now with proper newlines

PepperLola committed:
fix: better screen conditional rendering for configure, toasts render properly with html or strings

PepperLola committed:
feat: can change color mode from theme editor

PepperLola committed:
refactor: should support hot refresh now?

PepperLola committed:
Merge remote-tracking branch 'origin/dev' into jwrigh/1903/ui-refactor-tmp

PepperLola committed:
chore: resolved lint+format issues

PepperLola committed:
chore: merged dev

PepperLola committed:
Merge remote-tracking branch 'origin/dev' into jwrigh/1903/ui-refactor-tmp

PepperLola committed:
fix: resolved some circular dependencies

PepperLola committed:
fix: restored old biome config

PepperLola committed:
chore: run formatter

PepperLola committed:
chore: fix lint errors

PepperLola committed:
fix: finished resolving circular dependencies

PepperLola committed:
fix: resolved build issues

PepperLola committed:
fix(test): app mount test works with ui refactor

PepperLola committed:
fix(test): fixed most build and test issues

PepperLola committed:
fix(test): refactored to fix circular dependencies so tests pass

PepperLola committed:
chore: ran formatter

PepperLola committed:
fix: properly remove object when deleted in configure panel

Dhruv-0-Arora reviewed (changes requested):
1. No way to close the settings modal unless you go to the theme editor and close using the theme editor.

2. When the InputSchemeSelector is visible and you click "Enter", it does not select Ernie as intended and instead throws an error. Same happens if you don't select anything and just click "Finish"
<img width="664" height="327" alt="Screenshot 2025-07-29 at 4 27 33 PM" src="https://github.com/user-attachments/assets/622c8a40-fde0-4009-a1c8-2fa876e28b92" />

3. Make the Blue team's color when choosing input scheme not the secondary color and instead just a dark blue.

4. Fields disappear after clicking "Finish"

rutmanz commented on fission/src/events/ConfigurationSavedEvent.ts:
```diff
@@ -6,11 +6,11 @@ export class ConfigurationSavedEvent extends Event {
     }
 
-    public static listen(func: (e: Event) => void) {
+    public static Listen(func: (e: Event) => void) {
```
bad

rutmanz commented:
I think onBeforeAccept and onAccept should be switched so the panel can exfiltrate date to its caller

AlexD717 reviewed (changes requested):
1) Can select text by dragging with your mouse in text boxes (the panel moves).
2) In Configure Assets tab, clicking on the same tab already selected changes the category
3) Configure Assets input tab, doesn't show all input scheme, instead shows one input scheme to configure
4) Many panels are to high to fit on one screen
5) No scoreboard for match mode
6) Initially spawning in a robot, their name tag doesn't match the alliance (its grey)

PepperLola committed:
fix: swapped onAccept and onBeforeAccept

PepperLola committed:
fix: can no longer deselect mode in config panel

PepperLola committed:
fix: include default props that aren't overwritten

rutmanz commented on fission/src/systems/match_mode/MatchMode.ts:
```diff
@@ -110,16 +106,21 @@ class MatchMode {
     }
 
-    start(openModal: (modalName: string) => void) {
+    start(openModal: OpenModalFn) {
```
what if you used the global function here

PepperLola committed:
fix: can import fields now, made accordion expand icons bigger

PepperLola committed:
chore: ran formatter

PepperLola committed:
fix: set robot alliance in initial config panel, make station buttons stack in row

PepperLola commented:
> 1. No way to close the settings modal unless you go to the theme editor and close using the theme editor.
> 
>     2. When the InputSchemeSelector is visible and you click "Enter", it does not select Ernie as intended and instead throws an error. Same happens if you don't select anything and just click "Finish"
> 
> 
> <img alt="Screenshot 2025-07-29 at 4 27 33 PM" width="664" height="327" src="https://private-user-images.githubusercontent.com/100000915/472236358-622c8a40-fde0-4009-a1c8-2fa876e28b92.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM4NDg1MjUsIm5iZiI6MTc1Mzg0ODIyNSwicGF0aCI6Ii8xMDAwMDA5MTUvNDcyMjM2MzU4LTYyMmM4YTQwLWZkZTAtNDAwOS1hMWM4LTJmYTg3NmUyOGI5Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDczMFQwNDAzNDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZTMyNDA3NTE3MGFmMzhhY2Y5ZWU5NTEwNzVmM2Y3MjM4ZjgwNTNiMDQ5M2NiNDZjMmFkOGZjYjA3NWQ2MTk2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.XbdMF2NRo0nuzCuUFf9VTSgwYXBkVzfq_m4GTo0BfAo">
> 
>     3. Make the Blue team's color when choosing input scheme not the secondary color and instead just a dark blue.
> 
>     4. Fields disappear after clicking "Finish"

1. Fixed - non-overwritten default properties were being discarded when opening modals and panels
2. This error was related to 4, which was also fixed. I'm not sure how you're having behavior when pressing enter because that doesn't do anything for me
3. As discussed earlier, those buttons were a theme color just to signify that they weren't active, they weren't associated with the blue team. We might want to go in later and adjust the red and blue alliance colors to not be #ff0000 and #0000ff but for now I think it's fine
4. Fixed, the initial config panel was trying to close itself in a callback after the UI provider was closing it

PepperLola commented:
> I think onBeforeAccept and onAccept should be switched so the panel can exfiltrate date to its caller

I switched them and updated the PR description, let me know if it looks good and makes sense.

PepperLola commented:
> 1. Can select text by dragging with your mouse in text boxes (the panel moves).
> 
>     2. In Configure Assets tab, clicking on the same tab already selected changes the category
> 
>     3. Configure Assets input tab, doesn't show all input scheme, instead shows one input scheme to configure
> 
>     4. Many panels are to high to fit on one screen
> 
>     5. No scoreboard for match mode
> 
>     6. Initially spawning in a robot, their name tag doesn't match the alliance (its grey)

1. Fixed by adding "input" to selectors that aren't draggable
2. Fixed by not allowing null selections within the toggle button group
3. I don't have this issue. I'm not sure if I changed something that inadvertently fixed it but I see all the input schemes
4. I'm not sure this is within the scope of this PR - a different ticket is to improve the appearance of the UI which I think this makes more sense for; this PR was just to change the way we handle the UI and reimplement all the screens
5. Done
6. Fixed, the extra configuration in the initial config panel didn't make it through the merge :(

PepperLola committed:
feat: add back scoreboard

PepperLola committed:
feat: add theme reset button and fixed colors of modal buttons

rutmanz commented on fission/src/systems/simulation/SimConfigShared.ts:
```diff
@@ -258,14 +253,14 @@ export class SimConfig {
 
-    public static refreshRobotIO(config: SimConfigData) {
-        SimConfig.addRobotIONode(config)
+    public static RefreshRobotIO(config: SimConfigData) {
```
uh oh

itspvty reviewed (dismissed):
Could we conform this PR's description to the project's standard?

rutmanz commented on fission/src/ui/components/simulation/FlowControls.tsx:
```diff
@@ -1,21 +1,15 @@
-    )
+function FlowControlButton({ onClick, children }: PropsWithChildren<{ onClick?: () => void }>) {
+    return <Button onClick={() => onClick?.()}>{children}</Button>
 }
```
isn't this just actuall a Button?

rutmanz commented on fission/src/Synthesis.tsx:
```diff
@@ -107,40 +44,30 @@ const Synthesis: React.FC = () => {
+    // if (PreferencesSystem.getGlobalPreference("RenderScoreboard") && !scoreboardExists) {
+    //     openPanel("scoreboard")
+    // }
+    // });
```
why no scoreboard?

AlexD717 commented:
To me, it looks like the UI Refactor has gotten to a point, where multiple people can now work on it, and improve it. There are a lot of bugs that have to be fixed before this can be put into production, but I am fine with this being in dev, and a bunch of Jira tickets being made to fix all the issues.

Here are some more issues that I found (I think its fine to merge this first and fix these later)
1) Configure Assets input tab, doesn't show all input scheme, instead shows one input scheme to configure. - This only happens if you have at least one robot spawned
2) Right clicking on a robot and pressing move opens the configure assets panel, but doesn't place you in the correct spot.
3) End of match scoreboard UI is broken, but AARD-1896 already exists
4) Configure ejector tab has a unlabeled button (reset)
5) There appears to be no way to delete custom input schemes - in the assembly setup panel
6) There is no way to cancel creating a input scheme, on the model where you input the input scheme name
7) Drag mode arrow key rotation is extremely slow
8) Starting drag mode previously spawned a button in the bottom left that could be clicked to disable drag mode
9) You can select and highlight the title on the panels/modals
10) The checkbox/toggle-switch for persistent points in scoring zones doesn't change graphically but leaving the menu, and going back in updates it
11) The UI for scoring/protected zone config looks ugly
<img width="364" height="308" alt="Screenshot 2025-07-30 at 9 07 07 AM" src="https://github.com/user-attachments/assets/1e15f5b3-eba9-44e7-9118-4a57ecab18b7" />

12) Some dropdowns, when pressed remove part of the border
<img width="279" height="74" alt="Screenshot 2025-07-30 at 9 07 21 AM" src="https://github.com/user-attachments/assets/f76f07f5-61aa-4cc1-9ab9-43118c26e7fd" />

Personal Preference
- I think its very annoying how the spawn assets and configure assets spawn in the middle of the screen, since the robot you are spawning/configuring is usually in the middle of the screen, and so you can't see it.
- Instead of allowing the user to spawn multiple panels, if one is already open I think we should just bring that one up

azaleacolburn reviewed (dismissed):
Clicking an action item in the control panel for an object (E.g., `Move`) just opens the general Configure Assets Panel.

rutmanz commented on fission/src/ui/components/Modal.tsx:
```diff
@@ -1,158 +1,68 @@
-                            />
-                        )}
-                    </div>
+                    </CardActions>
```
I think `=== false` would make a lot more sense for these

rutmanz commented on fission/src/ui/components/Scoreboard.tsx:
```diff
@@ -0,0 +1,78 @@
+                    <Stack
+                        direction="column"
+                        className="items-center justify-center w-20 h-20 rounded-lg"
+                        sx={{ background: "#ff0000" }}
```
I feel like alliance red and blue should be part of the theme (in the MUI sense) or at least not defined everywhere they're referenced (and not these colors please my eyes)

rutmanz commented on fission/src/ui/components/Skybox.tsx:
```diff
@@ -1,10 +1,10 @@
+    // const { currentTheme, themes } = useTheme()
     if (World.sceneRenderer) {
-        World.sceneRenderer.updateSkyboxColors(themes[currentTheme])
+        // World.SceneRenderer.UpdateSkyboxColors(themes[currentTheme])
```
?

rutmanz commented on fission/src/ui/components/StatefulSlider.tsx:
```diff
@@ -0,0 +1,33 @@
+        tooltip?: string
+    }
+> = props => {
+    const [value, setValue] = useState(props.defaultValue)
```
This is the opposite of what StatefulCheckbox does (having an internal state variable). I called StatefulCheckbox that in contrast to the established Checkbox which held the state internally

PepperLola replied:
I renamed StatefulCheckbox to Checkbox since it doesn't hold state internally. I forget if that was what we settled on or if I got it backwards though.

rutmanz commented on fission/src/ui/helpers/StateProviderHelpers.ts:
```diff
@@ -0,0 +1,40 @@
+    configurePanelSettings?: ConfigurePanelSettings
+    setConfigurePanelSettings: (_settings?: ConfigurePanelSettings) => void
+    configurationType: ConfigurationType
+    setConfigurationType: (_type: ConfigurationType) => void
```
I kind of thought the point of being able to pass things back and forth between panels and the things that open/close them was to avoid this global state stuff

PepperLola replied:
The configure panel settings were only used in one place so I was able to make those a state just in that component. Otherwise, these are all pretty inherently global state– they're configured and used in screens that don't open each other and are settings that need to persist beyond the lifetime of any one panel. I can't think of a better way to manage it than a context like this, and I think it's a lot better than our solution before of having a shared object between multiple files.

BrandonPacewic reviewed (commented):
Two things:

1) Are we planning on pushing this before we make any more changes to the UI that would cause conflicts?
2) If we are, what's the plan for getting everything into a "polished" state before the end of the summer?

If we choose to wait then we are going to be doing two things of development, on dev and copying the changes over to this branch as development continues.

If we choose to merge this now then it looks like we have a lot of changes to make before we can release it and if so then how is that going to affect the rest of the stuff we have planned for the end of summer.

rutmanz commented:
These might not all go in this PR but things that I noticed
- Drag mode indicator is gone
- Toasts and graphics settings show up on the left now which overlaps more with main hud
- Can't dismiss toasts
- 
<details>
<summary>Tall things don't scroll</summary>

<img width="5088" height="3926" alt="image" src="https://github.com/user-attachments/assets/2898d7d8-3db3-4496-b598-eb825da380fc" />

</details>

<details>
<summary>Analytics consent has no background</summary>

<img width="586" height="331" alt="image" src="https://github.com/user-attachments/assets/faea4952-270a-42c1-a72a-439970aadbb0" />


</details>

<details>
<summary>Scoreboard doesn't respect preference for whether or not to show it and also the colors are blinding me</summary>

<img width="220" height="128" alt="image" src="https://github.com/user-attachments/assets/6abc4fef-2128-4902-891b-563f1ed6ddb6" />


</details>

<details>
<summary>Can't upload files (no continue button)</summary>

<img width="652" height="549" alt="image" src="https://github.com/user-attachments/assets/a4a62343-0b83-4ea4-bf67-bfa76a92fff4" />


</details>

<details>
<summary>Random deprication warning</summary>

<img width="1345" height="135" alt="image" src="https://github.com/user-attachments/assets/93d40590-f2ca-40b6-8649-58627284d14b" />


</details>

<details>
<summary>Why is it so wide?</summary>

<img width="919" height="537" alt="image" src="https://github.com/user-attachments/assets/84cfa843-5278-40d8-91bc-8c0aab7af6be" />


</details>

<details>
<summary>Dropdown multiselect isn't super clear and also doesn't move with the panel and also has a random missing border section</summary>

<img width="423" height="394" alt="image" src="https://github.com/user-attachments/assets/098eceab-62a1-45c2-8432-8131771f4bc9" />


</details>

<details>
<summary>Nametag text is dark</summary>

<img width="423" height="394" alt="image" src="https://github.com/user-attachments/assets/10df26b9-aa7a-4793-a61f-4699150400bd" />


</details>

AlexD717 commented:
> Nametag text is dark
> <img alt="image" width="423" height="394" src="https://private-user-images.githubusercontent.com/92497727/472633602-10df26b9-aa7a-4793-a61f-4699150400bd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM4OTcxMzUsIm5iZiI6MTc1Mzg5NjgzNSwicGF0aCI6Ii85MjQ5NzcyNy80NzI2MzM2MDItMTBkZjI2YjktYWE3YS00NzkzLWE2MWYtNDY5OTE1MDQwMGJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzMwVDE3MzM1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4ZmM5NDE3Mzc0NjAzZGE1NDg0NDg1NzIyZmQwYmM1MDg0OTdiMzc0NDUzZTUzZjA0ZTA0NTVhZGEyOWE2NGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8zRQwOZomyEiDFpTptNzqeVWQ8c9Ulhappasa8KSO8A">

It looks like the UI changes based on if the user has dark mode or light mode. Is that something that we want to support?

PepperLola committed:
fix: misc fixes

PepperLola committed:
fix: more misc fixes

PepperLola committed:
fix: more suggested fixes

PepperLola committed:
fix: use global open modal function for match mode

PepperLola committed:
fix: eliminate redundant custom flow button component

PepperLola committed:
feat: add red and blue alliance colors to mui theme

PepperLola commented:
> 1. Configure Assets input tab, doesn't show all input scheme, instead shows one input scheme to configure. - This only happens if you have at least one robot spawned
> 2. Right clicking on a robot and pressing move opens the configure assets panel, but doesn't place you in the correct spot.
> 3. End of match scoreboard UI is broken, but [AARD-1896](https://jira.autodesk.com/browse/AARD-1896) already exists
> 4. Configure ejector tab has a unlabeled button (reset)
> 5. There appears to be no way to delete custom input schemes - in the assembly setup panel
> 6. There is no way to cancel creating a input scheme, on the model where you input the input scheme name
> 7. Drag mode arrow key rotation is extremely slow
> 8. Starting drag mode previously spawned a button in the bottom left that could be clicked to disable drag mode
> 9. You can select and highlight the title on the panels/modals
> 10. The checkbox/toggle-switch for persistent points in scoring zones doesn't change graphically but leaving the menu, and going back in updates it
> 11. The UI for scoring/protected zone config looks ugly

1. This is an issue in dev and I don't think it's within the scope of this PR
2. In progress
3. I was expecting that task to take care of this
4. Fixed
5. Also an issue in dev - I created a Jira ticket to fix
6. Also an issue in dev - Jira ticket created
7. Also an issue in dev
8. Fixed
9. Fixed
10. Fixed
11. Not within the scope of this PR - another ticket to fix the UI appearance already exists

PepperLola committed:
fix: change panel open locations

PepperLola commented:
> * Drag mode indicator is gone
> * Toasts and graphics settings show up on the left now which overlaps more with main hud
> * Can't dismiss toasts

1. Drag mode indicator is back!!!!
2. Toasts now show up on the right, default panel spawn location is the right and the graphics panel opens in the middle
3. Can now dismiss toasts

> Tall things don't scroll
> Analytics consent has no background
> Scoreboard doesn't respect preference for whether or not to show it and also the colors are blinding me
> Can't upload files (no continue button)
> Random deprication warning
> Why is it so wide?
> Dropdown multiselect isn't super clear and also doesn't move with the panel and also has a random missing border section
> Nametag text is dark

1. I think this is not within the scope of this PR, but rather the one to fix the UI that will happen next
2. Fixed
3. Fixed
4. Fixed
5. I'm not sure why that's showing up on line 18 of Scoreboard.tsx since that just calls `useState`. However, a similar error appears for panels and originates from the react-draggable library.
6. Fixed (made it tall instead)
7. Fixed labels so there's text in the gap, added checkboxes to multi-select dropdown
8. Made nametag text always light

AlexD717 reviewed (changes requested):
Just a couple more issues that I found

1) If go into the configure score zone tab in configure assets, and then press the robot tab the app crashes

https://github.com/user-attachments/assets/889e64bf-bcf4-442b-849b-5dc2fac2326f

2) The scoring zone in configure assets panel has two buttons that both navigate back
<img width="397" height="413" alt="Screenshot 2025-07-31 at 8 28 50 AM" src="https://github.com/user-attachments/assets/4f0bee7e-319e-4d0a-ac41-f0d456521086" />

AlexD717 commented:
Also:

3) In the intake configure tab, show intake zone indicator always, doesn't graphically update
4) Is there a way to have the slider show what value they are on, because this isn't very helpful
<img width="293" height="61" alt="Screenshot 2025-07-31 at 9 10 09 AM" src="https://github.com/user-attachments/assets/10724d55-5090-4ede-a9b7-ad3a80b9a893" />

ryanzhangofficial commented:
Some suggestions. These might part of 2002 though.
1. The buttons in debug tools, developer tool, and match mode config seem jammed together and uncomfortable to look at it. Perhaps make the borders the bright blue and the body the same color as the background?
2. Perhaps make the saved robot text the same size as the 6 default robots so it's easy to distinguish between section title (Saved Robot) and the actual saved robots.
3. I agree with Alexey on his point on having slider show what value they are on. Other examples are the ejector velocity,  I think this should be standardized throughout fission.

Bugs:
1. The issue resolved in [AARD-1996](https://github.com/Autodesk/synthesis/pull/1218) have resurfaced.
2. Receiving the error below when you add click a field -> click on scoring zones or protected zones -> click on Robot tab.
<img width="476" height="703" alt="Screenshot 2025-07-31 094405" src="https://github.com/user-attachments/assets/ad3574c6-6ad1-456f-a85e-46fbf04722f6" />

ryanzhangofficial reviewed (changes requested):
Some suggestions:
1. The buttons in debug tools, developer tool, and match mode config seem jammed together and uncomfortable to look at it. Perhaps make the borders the bright blue and the body the same color as the background?
2. Perhaps make the saved robot text the same size as the 6 default robots so it's easy to distinguish between section title (Saved Robot) and the actual saved robots.
3. I agree with Alexey on his point on having slider show what the actual value is. Another example is the ejector velocity which had a number before as well. I think this should be standardized throughout fission.

Some bugs:
1. The issue resolved in [AARD-1996](https://github.com/Autodesk/synthesis/pull/1218) has resurfaced.
2. Receiving the errors pictured below when I click on a field -> click scoring or protected zones -> click on robot. Fission crashes.

<img width="476" height="703" alt="Image" src="https://github.com/user-attachments/assets/b20b9d3b-6c0f-438f-b5f2-c01f13e78ce7" />

PepperLola committed:
fix: switching to robot tab no longer crashes fission

PepperLola committed:
fix: always show intake switch updates properly

PepperLola commented:
> Just a couple more issues that I found
> 
> 1. If go into the configure score zone tab in configure assets, and then press the robot tab the app crashes
> 
> 2. The scoring zone in configure assets panel has two buttons that both navigate back

Both fixed

PepperLola commented:
> Also:
> 
> 3. In the intake configure tab, show intake zone indicator always, doesn't graphically update
> 4. Is there a way to have the slider show what value they are on, because this isn't very helpful

3. Fixed
4. I don't think that's within the scope of this PR, can be done in the UI appearance improvement one. I agree with the change though

PepperLola committed:
chore: removed configure panel settings from global state and fixed color sx props

PepperLola committed:
fix: removed path prefix from robot names when downloading all

PepperLola committed:
fix: next button appears properly with local mira import

PepperLola commented:
> Some suggestions:
> 
> 1. The buttons in debug tools, developer tool, and match mode config seem jammed together and uncomfortable to look at it. Perhaps make the borders the bright blue and the body the same color as the background?
> 2. Perhaps make the saved robot text the same size as the 6 default robots so it's easy to distinguish between section title (Saved Robot) and the actual saved robots.
> 3. I agree with Alexey on his point on having slider show what the actual value is. Another example is the ejector velocity which had a number before as well. I think this should be standardized throughout fission.

I don't think these fall within the scope of this PR– I definitely agree with the feedback but I think that will be covered by the UI improvement PR later.

> 
> Some bugs:
> 
> 1. The issue resolved in [AARD-1996](https://github.com/Autodesk/synthesis/pull/1218) has resurfaced.
> 2. Receiving the errors pictured below when I click on a field -> click scoring or protected zones -> click on robot. Fission crashes.

Both have been fixed

PepperLola committed:
fix: context menu opens correct configure panel option

PepperLola committed:
feat: add custom props to screens

PepperLola committed:
Merge remote-tracking branch 'origin/dev' into jwrigh/1903/ui-refactor-tmp

PepperLola committed:
fix: formatter

PepperLola committed:
fix: updated dependencies

trying to fix unit tests failing in CI

AlexD717 reviewed (changes requested):
If you have a robot and a field spawned in, and you right click on the field and press configure the app crashes.

PepperLola committed:
fix: removed configurationType from global state

instead pass as prop just like configType and selectedAssembly - not
sure why it wasn't that way before. fixes field configuration issue

Dhruv-0-Arora reviewed (changes requested):
Graphics settings panel should be more descriptive. I think its better to just make this another PR though and get the UI Refactor merged.

<img width="250" height="613" alt="Screenshot 2025-08-04 at 4 03 19 PM" src="https://github.com/user-attachments/assets/874e9ae5-dc9e-4374-a313-c8556d87e81a" />

This is what graphics settings looks like on dev

<img width="436" height="461" alt="Screenshot 2025-08-04 at 4 07 07 PM" src="https://github.com/user-attachments/assets/9a3ac6d1-7e41-44f8-ac65-cbbc44400640" />


[I realized that its not to do with th  graphics settings panel but how the sliders are formatted. The value of the slider is not displayed which, in my opinion, would be useful be to add]

Dhruv-0-Arora commented:
Potential feature to be added in the future: 

Save the positioning of panels in preferences so that they always open in the position set by the user.

rutmanz reviewed (approved):
I think all of the core functionality seems good, and it's just style things remaining

ryanzhangofficial reviewed (approved):
If the criteria for merging is no crashing and retaining functionality, I think we're good to go.

PepperLola commented:
> Graphics settings panel should be more descriptive. I think its better to just make this another PR though and get the UI Refactor merged.
> 
> 
> 
> <img width="250" height="613" alt="Screenshot 2025-08-04 at 4 03 19 PM" src="https://github.com/user-attachments/assets/874e9ae5-dc9e-4374-a313-c8556d87e81a" />
> 
> 
> 
> This is what graphics settings looks like on dev
> 
> 
> 
> <img width="436" height="461" alt="Screenshot 2025-08-04 at 4 07 07 PM" src="https://github.com/user-attachments/assets/9a3ac6d1-7e41-44f8-ac65-cbbc44400640" />
> 
> 
> 
> 
> 
> [I realized that its not to do with th  graphics settings panel but how the sliders are formatted. The value of the slider is not displayed which, in my opinion, would be useful be to add]
> 
> 

These are style changes that are not within the scope of this PR

PepperLola committed:
fix(build): resolved ArrayBuffer vs Uint8Array<ArrayBufferLike> type issues

I didn't really think through this solution much so it might need to be
revisited at some point. It fixes the build though, and it looks like it
didn't affect any mirabuf parsing

Dhruv-0-Arora reviewed (approved):
This is a work of ART

AlexD717 reviewed (approved):
Lets get this in!

As a different ticket, we should consider removing a lot of the console.log statements, as it currently clutters up the console.

BrandonPacewic reviewed (approved):
@PepperLola I'll let you be the final review and merge for this PR. From what we discussed yesterday in standup everyone is ready to get this in and begin working on it. Plus we can finally merge everything else from the backlog.

PepperLola committed:
Merge remote-tracking branch 'origin/dev' into jwrigh/1903/ui-refactor-tmp
